<?xml version='1.0' encoding='UTF-8'?>
<root version="2.3.8">
 <tabbed_widget name="Main Window" parent="main_window">
  <plotmatrix rows="2" columns="3" tab_name="imu_data">
   <plot row="0" style="Lines" col="0">
    <range bottom="-1.686117" top="1.556830" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="76" custom_transform="noTransform" B="193" name="/uav68/mavros/imu/data/linear_acceleration/x" R="241"/>
    <curve G="119" custom_transform="noTransform" B="180" name="/uav68/os1_cloud_node/imu/linear_acceleration/x" R="31"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/linear_acceleration/x" R="0"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="1" style="Lines" col="0">
    <range bottom="-0.188712" top="0.298966" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="189" custom_transform="noTransform" B="34" name="/uav68/mavros/imu/data/angular_velocity/x" R="188"/>
    <curve G="127" custom_transform="noTransform" B="14" name="/uav68/os1_cloud_node/imu/angular_velocity/x" R="255"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/angular_velocity/x" R="0"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="1">
    <range bottom="-0.939724" top="1.322796" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="103" custom_transform="noTransform" B="189" name="/uav68/mavros/imu/data/linear_acceleration/y" R="148"/>
    <curve G="190" custom_transform="noTransform" B="207" name="/uav68/os1_cloud_node/imu/linear_acceleration/y" R="23"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/linear_acceleration/y" R="0"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="1" style="Lines" col="1">
    <range bottom="-0.332539" top="0.262098" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="119" custom_transform="noTransform" B="180" name="/uav68/mavros/imu/data/angular_velocity/y" R="31"/>
    <curve G="39" custom_transform="noTransform" B="40" name="/uav68/os1_cloud_node/imu/angular_velocity/y" R="214"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/angular_velocity/y" R="0"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="2">
    <range bottom="8.579801" top="11.035893" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="190" custom_transform="noTransform" B="207" name="/uav68/mavros/imu/data/linear_acceleration/z" R="23"/>
    <curve G="76" custom_transform="noTransform" B="193" name="/uav68/os1_cloud_node/imu/linear_acceleration/z" R="241"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/linear_acceleration/z" R="0"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="1" style="Lines" col="2">
    <range bottom="-0.108287" top="0.334789" right="296.629149" left="0.000000"/>
    <limitY/>
    <curve G="39" custom_transform="noTransform" B="40" name="/uav68/mavros/imu/data/angular_velocity/z" R="214"/>
    <curve G="189" custom_transform="noTransform" B="34" name="/uav68/os1_cloud_node/imu/angular_velocity/z" R="188"/>
    <curve G="0" custom_transform="noTransform" B="0" name="/uav68/os1_cloud_node/imu_filt/angular_velocity/z" R="0"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <plotmatrix rows="1" columns="3" tab_name="orientation">
   <plot row="0" style="Lines" col="0">
    <range bottom="-2.488035" top="8.394711" right="218.026167" left="0.000000"/>
    <limitY/>
    <curve G="190" custom_transform="noTransform" B="207" name="/uav68/mavros/imu/data/orientation/roll_deg" R="23"/>
    <curve G="76" custom_transform="noTransform" B="193" name="/uav68/os1_cloud_node/imu_atti/orientation/roll_deg" R="241"/>
    <transform value="Custom::noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="1">
    <range bottom="-8.666466" top="4.167489" right="218.026167" left="0.000000"/>
    <limitY/>
    <curve G="189" custom_transform="noTransform" B="34" name="/uav68/mavros/imu/data/orientation/pitch_deg" R="188"/>
    <curve G="103" custom_transform="noTransform" B="189" name="/uav68/os1_cloud_node/imu_atti/orientation/pitch_deg" R="148"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="2">
    <range bottom="-188.258271" top="188.346125" right="218.026167" left="0.000000"/>
    <limitY/>
    <curve G="119" custom_transform="noTransform" B="180" name="/uav68/mavros/imu/data/orientation/yaw_deg" R="31"/>
    <curve G="190" custom_transform="noTransform" B="207" name="/uav68/os1_cloud_node/imu_atti/orientation/yaw_deg" R="23"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <plotmatrix rows="1" columns="3" tab_name="plot">
   <plot row="0" style="Lines" col="0">
    <range bottom="-166.884638" top="-73.487814" right="217.897550" left="212.915333"/>
    <limitY/>
    <curve G="103" custom_transform="noTransform" B="189" name="/lio_sam/mapping/odometry/pose/position/x" R="148"/>
    <curve G="39" custom_transform="noTransform" B="40" name="/uav68/lio_sam/odometry/imu/pose/position/x" R="214"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="1">
    <range bottom="140.597606" top="230.136174" right="217.897550" left="212.915333"/>
    <limitY/>
    <curve G="190" custom_transform="noTransform" B="207" name="/lio_sam/mapping/odometry/pose/position/y" R="23"/>
    <curve G="201" custom_transform="noTransform" B="56" name="/uav68/lio_sam/odometry/imu/pose/position/y" R="26"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="2">
    <range bottom="-457.726935" top="-382.752679" right="217.897550" left="212.915333"/>
    <limitY/>
    <curve G="119" custom_transform="noTransform" B="180" name="/lio_sam/mapping/odometry/pose/position/z" R="31"/>
    <curve G="127" custom_transform="noTransform" B="14" name="/uav68/lio_sam/odometry/imu/pose/position/z" R="255"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <currentPlotMatrix index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="ROS /rosout Visualization" status="idle"/>
  <plugin ID="ROS Topic Re-Publisher" status="idle"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets>
  <snippet language="JS" name="1st_derivative">
   <global>var prevX = 0
var prevY = 0</global>
   <equation>dx = time - prevX
dy = value - prevY
prevX = time
prevY = value

return dy/dx</equation>
  </snippet>
  <snippet language="JS" name="1st_order_lowpass">
   <global>var prevY = 0
var alpha = 0.1</global>
   <equation>prevY = alpha * value + (1.-alpha) * prevY

return prevY</equation>
  </snippet>
  <snippet language="JS" name="2D_velocity">
   <global>var prev_x = 0
var prev_y = 0
var prev_t = 0</global>
   <equation>X = $$your_odometry/position/x$$
Y = $$your_odometry/position/y$$

var dist = sqrt( (X-prev_x)*(X-prev_x) + (Y-prev_y)*(Y-prev_y) )
var dT = time - prev_t

prev_x = X
prev_y = Y
prev_t = time

return dist / dT</equation>
  </snippet>
  <snippet language="JS" name="average_two_curves">
   <global></global>
   <equation>a = $$PLOT_A$$
b = $$PLOT_B$$

return (a+b)/2</equation>
  </snippet>
  <snippet language="JS" name="integral">
   <global>var integral = 0</global>
   <equation>integral += value
return integral</equation>
  </snippet>
  <snippet language="JS" name="rad_to_deg">
   <global></global>
   <equation>return value*180/3.1417</equation>
  </snippet>
  <snippet language="JS" name="remove_offset">
   <global>var is_first = true
var first_value = 0</global>
   <equation>if (is_first)
{
  is_first = false
  first_value = value
}

return value - first_value</equation>
  </snippet>
  <snippet language="JS" name="yaw_from_quaternion">
   <global>// source: https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles

function quaternionToYaw(x, y, z, w)
{
  // yaw (z-axis rotation)
  t1 = 2.0 * (w * z + x * y);
  t2 = 1.0 - 2.0 * (y * y + z * z);
  yaw = Math.atan2(t1, t2);

  return yaw
}</global>
   <equation>return quaternionToYaw(x, y, z, w);</equation>
  </snippet>
 </snippets>
 <!-- - - - - - - - - - - - - - - -->
</root>

